File: Perl/odbc-test.pl
use DBI;
# DSN (Data Source Name) must match the name configured in your odbc.ini or ODBC Administrator
my $dsn = "dbi:MariaDB:database=luciano;host=192.168.1.181"; 
my $user = "tenchi";
my $password = "xxx";

use v5.42;

my $dbh = DBI->connect($dsn, $user, $password) or die "Couldn't open database: $DBI::errstr\\n";


my $stmt = "select * from accounts";
my $sth = $dbh->prepare($stmt);
$sth->execute();

while(my @row = $sth->fetchrow_array()) {
   say @row;
}



$sth->finish();
$dbh->disconnect();

File: PHP/odbc-test.php
<?php
// $db = new PDO('odbc:Driver=FreeTDS;Server=host;Port=port;Database=database;UID=user;PWD=pass;');


$dsn = "odbc:Driver=FreeTDS;Server=vssqldmb01.bmcc.cuny.edu;Port=1433;Database=Tenchi;"; // DSN_NAME is the name of your configured DSN
$user = "tenchi";
$password = "xxx";
$options = [
    PDO::ATTR_ERRMODE            => PDO::ERRMODE_EXCEPTION, // Throw exceptions on errors
    PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC,       // Default to associative arrays
    PDO::ATTR_EMULATE_PREPARES   => false,                  // Use native prepared statements
];
try {
    $conn = new PDO($dsn, $user, $password, $options);
    // $conn = new PDO($dsn, $user, $password);
    echo "Connection successful!";
  
    $stmt = $conn->query("SELECT name FROM transcripts");
    while ($row = $stmt->fetch()) {
        echo $row['name'] . "\n";
    }



} catch (PDOException $e) {
    die("Unable to connect: " . $e->getMessage());
}


?>


